wordpress-mobile/WordPress-Android
# Feature/whats new v2

This PR adds support for remote announcements. 

Flux-C counterpart is [here](https://github.com/wordpress-mobile/WordPress-FluxC-Android/pull/1605)

Remote announcements introduced a minor change of logic - we only show announcement on app upgrade if it's available in the cache since we can't predict the timing of announcement arrival when requested from the backend. This means, that on app upgrade, you load the announcement details, and on subsequent app launch you will see the announcement details (unless you checked the announcement in App Settings)

To test:
Testing -1  - 13.8 bracket:
- Set version name to 13.7, and do a clean app install.
- Log in and navigate to the My Site screen. 
- You should not see the announcement on app upgrade.
- Don’t check the announcement in App Settings.
- Close the app.
- Bump version name to 13.8 and bump versionCode
- Launch the app. You should see the announcement on app upgrade, since it’s already cached.

Testing that announcement will not pop up on app upgrade if it was checked in App Settings.
- Set version name to 13.7, and do a clean app install.
- Login and navigate to the My Site screen.
- Go to app settings and confirm that the feature announcement is available at the bottom of the list. If your internet is really slow you might need to wait for it to be fetched from the endpoint. 
- Open the feature announcement.
- Close the app.
- Bump version name to 13.8 and bump versionCode.
- Launch the app. You should not see the announcement on app upgrade, since you already saw it from App Settings.

Testing version with no announcement.
- Bump version name to 13.9 and bump versionCode.
- Start and then relaunch the app.
- You should not see the announcement, and the announcement should not be available in App Settings.

Testing a single version target.
- Bump version name to 14.0 and bump versionCode. 
- Start and then relaunch the app. 
- You should see the announcement that targets 14.0.

Testing 14.1-14.2 bracket.
- Bump version name to 14.1 and bump versionCode. 
- Start and then relaunch the app. You should see the announcement that targets 14.1

Testing 14.2-14.3 bracket.
- Bump version name to 14.2 and bump versionCode. 
- Start and then relaunch the app. You should see the announcement that targets 14.2

Testing 14.5  - - 1 bracket:
- Bump version name to 15.0 and bump versionCode. Start and then relaunch the app. 
- You should see the announcement that targets 14.5

PR submission checklist:
